src/plugins/intel_cpu/src/emitters/snippets/aarch64/utils.cpp

```diff
@@ -154,28 +153,52 @@ void push_and_load_ptrs_with_offsets(dnnl::impl::cpu::aarch64::jit_generator* h,
     const auto sp_size = rnd_up(mem_ptrs.size() * gpr_length, sp_alignment);
     h->sub(h->sp, h->sp, sp_size);
 
-    // Push all pointers with offsets onto stack
+    // Generate stack offsets for sequential storage
+    std::vector<size_t> stack_offsets;
+    stack_offsets.reserve(mem_ptrs.size());
+    for (size_t i = 0; i < mem_ptrs.size(); i++) {
+        stack_offsets.push_back(i * gpr_length);
+    }
+
+    // Use the common function to push pointers with offsets to stack
+    push_ptrs_with_offsets_to_stack(h, mem_ptrs, memory_offsets, buffer_ids, aux_regs, stack_offsets);
+
+    // Load back the adjusted pointers to specified registers
+    for (size_t i = 0; i < load_regs.size() && i < mem_ptrs.size(); i++) {
```

Comment:
I think we should firstly check for `mem_ptrs.size() == mem_ptrs.size()` and throw exception if it's not right.
And then we can just have only one condition on condition block in `for` loop